### PR TITLE
Utility methods for measuring text

### PR DESCRIPTION
**Pull request by dnfield**
Now that `TextPainter`s must be disposed, it makes some sense to wrap this functionality in a utility method.

In customer applications, it is very common to use a text painter to figure out the width of something you want to animate, and then immediately dispose of the painter. This is probably ok because of caching that SkParagraph will do.

I'm a little bit on the fence about whether we should be promoting this, but there are at least 100 callsites in Google's monorepo that use this pattern (likely with some cargo-culting since the code is basically the same everywhere) - so if there's some better way I'm open to hearing about it :)

### CODE REVIEW

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/painting/text_painter.dart [hunk 1]
```diff
@@ -217,6 +217,86 @@ class TextPainter {
...
+  /// Computes the width of a configured [TextPainter].
+  ///
+  /// This is a convenience method that creates a text painter with the supplied
+  /// parameters, lays it out with the supplied [minWidth] and [maxWidth], and
+  /// returns its [TextPainter.width] making sure to dispose the underlying
+  /// resources.
+  static double computeWidth({
+    required InlineSpan text,
+    TextAlign textAlign = TextAlign.start,
+    TextDirection? textDirection,
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/painting/text_painter.dart:229]: TextDirection must be non null I think?

Comment (dnfield) [reply packages/flutter/lib/src/painting/text_painter.dart:229]: This was copied from the `TextPainter` ctor. I'd like to keep the types the same for ease of migration.

If we want to update this we should update the ctor as well.

Comment (dnfield) [reply packages/flutter/lib/src/painting/text_painter.dart:229]: Ahh but I see what you're saying, on the regular object we can set this to some value later, whereas we can't with this method. Ok. In that case, I'll file a follow upt o make it non-nullable.

**Pull request merged**